### ISSUE TITLE
Problem with DataLoader

### ISSUE THREAD

**Issue by suchithtuple** [pytorch/pytorch#20717]
## To Reproduce
I have training , validation and test dataset. I have created dataloaders for three of them with training shuffle = True and for valdiation and test shuffle = False . So in one of the code epoch loop contains iterator over train and validation and in another one , epoch loop contains iterator over train , validation and test.

Check the output of the code 1 and 2
1. https://www.kaggle.com/suchith0312/pytorch-dataloader-testing?scriptVersionId=14425205
2. https://www.kaggle.com/suchith0312/pytorch-dataloader-testing?scriptVersionId=14425182

## Expected behavior

The train id's from output of files must be same but they are different. 

## Environment
 - PyTorch Version : 1.0.1.post2
 - Python version: 3.6.6
 - CUDA version : 10.0
- Cudnn version 7.4.2

**Comment by SsnL** [comment]
Your iteration over test advances rng state.

**Comment by suchithtuple** [comment]
> Your iteration over test advances rng state.

How can I preventing that or there is no way to get the outputs of two codes same.

**Comment by InnovArul** [comment]
A possible solution could be: https://github.com/pytorch/pytorch/pull/20749
